Array_Vector_Exercises/Exercise1/Menu.cpp

```diff
@@ -0,0 +1,380 @@
+			int MenuInput;
+			std::cin >> MenuInput;
+
+			while (std::cin.fail() || MenuInput != 1)
+			{
+				std::cin.clear();
+				std::cin.ignore(100, '\n');
+				std::cout << "Invalid Option, please enter a valid Option " << std::endl;
+				std::cin >> MenuInput;
+				if (MenuInput == 1)
+				{
+					ActiveMenuCommand = EMenuCommand::DynamicArray;
+				}
+			}
+		}
+
+		else
+		{
+			//freeing Memory
+			for (int i = 0; i < Shapes.GetSize(); i++)
+			{
+				delete Shapes[i];
+			}
+			ActiveMenuCommand = EMenuCommand::MainMenu;
+		}
+
+	} while (UserInput != 3);
+}
+
+FShape* FMenu::CreateShape()
+{
+	std::cout << "-------------------------------------" << std::endl;
+	std::cout << "1- Circle" << std::endl;
+	std::cout << "2- Square" << std::endl;
+	std::cout << "-------------------------------------" << std::endl;
+
+	int UserInput;
+	UserInput = ValidateInput(UserInput, 1, 2);
+
+	if (UserInput == 1)
+	{
+		std::cout << "Insert circle's Radius: ";
+		float InRadius;
+		
+		InRadius = ValidateInputFloat(InRadius);
+		return new FCircle(InRadius);
+		
+			
+	}
+	else
+	{
+		std::cout << "Insert square's lenght: ";
+		float InLenght;
+		InLenght = ValidateInputFloat(InLenght);
+
+		std::cout << "Insert square's heignt: ";
+		float InHeight;
+		InHeight = ValidateInputFloat(InHeight);
+
+		return new FSquare(InLenght, InHeight);
+	}
+}
+
+//---------------------------------------------------------------------------------
+void FMenu::PrintArray(const FShape& Shape) const
+{
+	std::cout << "this Shape is a: " << Shape.GetKindOfShape() << std::endl;
+	std::cout << Shape.PrintParameters() << std::endl;
+	std::cout << "the Shape's area is: " << Shape.GetArea() << std::endl;
+	std::cout << "the Shape's perimeter is: " << Shape.GetPerimeter() << std::endl;
+	std::cout << "--------------------------------- " << std::endl;
+};
+
+//---------------------------------------------------------------------------------
+void FMenu::RunApp()
+{
+	while (ActiveMenuCommand != EMenuCommand::Exit)
+	{
+		switch (ActiveMenuCommand)
+		{
+		case EMenuCommand::MainMenu:
+			OpenMainMenu();
+			break;
+		case EMenuCommand::BuildInArray:
+			AskForBuildInArray();
+			break;
+		case EMenuCommand::StaticArray:
+			AskForStaticArray();
+			break;
+		case EMenuCommand::DynamicArray:
+			AskForDynamicArray();
+			break;
+		case EMenuCommand::Exit:
+			break;
+		}
+	}
+}
+
+//------------------------------------------------------------------------------
+int FMenu::ValidateInput(int& UserInput, const int LowerBounce, const int HigherBounce)
+{
+	std::cin >> UserInput;
+
+	while (std::cin.fail() || UserInput < LowerBounce || UserInput > HigherBounce)
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid Option, please enter a valid Option " << std::endl;
+		std::cin >> UserInput;
+	};
+	return UserInput;
+}
+
+//---------------------------------------------------------------------------------
+int FMenu::ValidateInput(int& UserInput)
```

Comment:
Same here as the previous comment. Also, remember that since this isn't really validating an input, but returning a valid input, this may be more appropriately named. Something like GetValidInput or something like that.